x/evidence/genesis.go

```diff
@@ -16,17 +19,34 @@ func InitGenesis(ctx sdk.Context, k keeper.Keeper, gs types.GenesisState) {
 	}
 
 	for _, e := range gs.Evidence {
-		if _, ok := k.GetEvidence(ctx, e.Hash()); ok {
-			panic(fmt.Sprintf("evidence with hash %s already exists", e.Hash()))
+		evi, ok := e.GetCachedValue().(exported.Evidence)
+		if !ok {
+			panic("expected evidence")
+		}
+		if _, ok := k.GetEvidence(ctx, evi.Hash()); ok {
+			panic(fmt.Sprintf("evidence with hash %s already exists", evi.Hash()))
 		}
 
-		k.SetEvidence(ctx, e)
+		k.SetEvidence(ctx, evi)
 	}
 }
 
 // ExportGenesis returns the evidence module's exported genesis.
 func ExportGenesis(ctx sdk.Context, k keeper.Keeper) types.GenesisState {
```

Comment:
can you add a test too? 🙏

Reply:
please check out https://github.com/cosmos/cosmos-sdk/blob/marie/5917-genesis-proto/x/evidence/genesis_test.go
for the same reason as exposed [here](https://github.com/cosmos/cosmos-sdk/pull/6593#discussion_r452844045), it seems hard to test https://github.com/cosmos/cosmos-sdk/blob/marie/5917-genesis-proto/x/evidence/genesis.go#L40-L46